base_layer/core/src/blocks/block_header.rs

```diff
@@ -230,6 +230,15 @@ impl BlockHeader {
             .into()
     }
 
+    pub fn merge_mining_hash(&self) -> Vec<u8> {
+        let mining_prefix = (b"TARI").to_vec();
+        let mut mining_hash = self.mining_hash().to_vec();
+        for i in 0..4 {
+            mining_hash[i] = mining_prefix[i];
+        }
+        mining_hash
+    }
+
```

Comment:
You can change the FixedHash byte content without allocations
```suggestion
    pub fn merge_mining_hash(&self) -> FixedHash {
     let mut mining_hash = self.mining_hash();
        mining_hash[0..4].copy_from_slice(b"TARI"); // Maybe put this in a `const`
        mining_hash
    }
```